ui/pages/confirmations/utils/confirm.test.ts

```diff
@@ -97,4 +99,25 @@ describe('confirm util', () => {
       expect(result).toStrictEqual(false);
     });
   });
+
+  describe('isValidASCIIURL', () => {
+    it('returns true for URL containing only ASCII characters', () => {
+      expect(isValidASCIIURL('https://www.google.com')).toEqual(true);
+    });
+
+    it('returns false for URL containing special character', () => {
+      expect(isValidASCIIURL('https://iոfura.io/gnosis')).toStrictEqual(false);
+    });
+  });
+
+  describe('toPunycodeURL', () => {
+    it('returns punycode version of URL', () => {
+      expect(toPunycodeURL('https://iոfura.io/gnosis')).toStrictEqual(
+        'https://xn--ifura-dig.io/gnosis',
+      );
+      expect(toPunycodeURL('https://www.google.com')).toStrictEqual(
+        'https://www.google.com/',
+      );
```

Comment:
I think it would be nice to add a test case to include other URL parts and demonstrate it may encode other characters e.g. 
```suggestion
      expect(
        toPunycodeURL('https://iոfura.io/gnosis:5050?test=iոfura&foo=bar'),
      ).toStrictEqual(
        'https://xn--ifura-dig.io/gnosis:5050?test=i%D5%B8fura&foo=bar',
      );
      expect(toPunycodeURL('https://www.google.com')).toStrictEqual(
        'https://www.google.com/',
      );
```

Reply:
done 👍